src/main/frontend/handler/repo.cljs

```diff
@@ -413,31 +359,12 @@
                        (println "Git push error: ")
                        (js/console.error error)
                        (common-handler/check-changed-files-status repo-url)
-                       (let [permission? (or (string/includes? (str error) "401")
-                                             (string/includes? (str error) "404"))]
-                         (cond
-                           (and permission? (not fallback?))
-                           (request-app-tokens!
-                            (fn []
-                              (git-handler/set-git-status! repo-url :re-push)
-                              (push repo-url
-                                    {:commit-message commit-message
-                                     :fallback? true}))
-                            nil)
-
-                           :else
-                           (do
-                             (git-handler/set-git-status! repo-url :push-failed)
-                             (git-handler/set-git-error! repo-url error)
-                             (cond
-                               permission?
-                               (show-install-error! repo-url (util/format "Failed to push to %s. " repo-url))
-
-                               (state/online?)
-                               (pull repo-url token {:force-pull? true})
-
-                               :else    ; offline
-                               nil)))))))))))
+                       (do
+                         (git-handler/set-git-status! repo-url :push-failed)
+                         (git-handler/set-git-error! repo-url error)
+                         (when
+                           (state/online?)
```

Comment:
We can move this line after the above `when`.

Reply:
what's the meaning of `online` ?

Reply:
I misunderstand your meaning, oops